x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsCollector.java

```diff
@@ -87,7 +87,7 @@ public ClusterStatsCollector(final Settings settings,
     @Override
     protected boolean shouldCollect(final boolean isElectedMaster) {
         // This collector can always collect data on the master node
-        return isElectedMaster;
+        return isElectedMaster && super.shouldCollect(isElectedMaster);
```

Comment:
@pickypg Can you verify that this change is correct? It's not strictly related to this PR but some changes in this PR caused a test for this method to fail which led me to make this change.

Reply:
As noted earlier, I think it would be better to just handle this by scheduling. The reason this one did not use `super` is so that it would collect even if the license expired.

Reply:
@pickypg I'm undoing this change because of the scheduler check you mentioned in the other comment. However, just for my education, I'd like to understand why this collector is special in that it should collect even if the license expired?

Reply:
It's special because it's the core collector that the rest of the UI depends upon to show anything. So by having this document, we could still show the license expired part of the UI for example.